### PR TITLE
DM-47673: Make development branch for v28 release

### PR DESCRIPTION
**Pull request by JeremyMcCormick**
## Checklist

- [ ] Ran Jenkins
- [ ] Added a release note for user-visible changes to `docs/changes`

### CODE REVIEW

File: docs/CHANGES.rst [hunk 1]
```diff
@@ -1,3 +1,26 @@
...
+------------
+
+- Added a new ``tap_schema_`` module designed to deprecate and eventually replace the ``tap`` module.
+  This module provides utilities for translating a Felis schema into a TAP_SCHEMA representation.
+  The command ``felis load-tap-schema`` can be used to activate this functionality. (`DM-45263 <https://jira.lsstcorp.org/browse/DM-45263>`_)
+- Added a check to the data model which ensures that all constraint names are unique within the schema.
+  TAP_SCHEMA uses these names as primary keys in its ``keys`` table, so they cannot be duplicated. (`DM-45623 <https://jira.lsstcorp.org/browse/DM-45623>`_)
+- Added automatic ID generation for objects in Felis schemas when the ``--id-generation`` flag is included on the command line.
+  This is supported for the ``create`` and ``validate`` commands. (`DM-45938 <https://jira.lsstcorp.org/browse/DM-45938>`_)
+- Added a Schema validator function that checks if index names are unique.
```
Comment (timj) [inline docs/CHANGES.rst:14]: No ticket for this one? I think it's DM-45938 as well.

**Review by timj — APPROVED** [review]

File: docs/CHANGES.rst (hunk 1 above)
Comment (JeremyMcCormick) [reply docs/CHANGES.rst:14]: I don't think there was a specific ticket for this. It was merged as part of DM-45938 but was an unrelated change.

Comment (JeremyMcCormick) [reply docs/CHANGES.rst:14]: It was merged from the DM-45938 branch but was unrelated to that ticket. (Probably it should have had its own ticket.)

Comment (JeremyMcCormick) [reply docs/CHANGES.rst:14]: I can also list it as DM-45938 and add a note to the ticket on Jira about the additional change.

Comment (timj) [reply docs/CHANGES.rst:14]: It was merged on that ticket branch so we need to list that ticket so people could work out where it came from.

Comment (JeremyMcCormick) [reply docs/CHANGES.rst]: I changed this to one bullet listing both changes with a link to the ticket to make it clearer.

Comment (JeremyMcCormick) [reply docs/CHANGES.rst]: This seems okay to me now:

https://felis.lsst.io/builds/592/CHANGES.html

File: docs/CHANGES.rst [hunk 2]
```diff
@@ -1,4 +1,27 @@
-Felis 27.0.0 (2024-04-17)
+28.0.0 (2024-11-26)
+==========================
```
Comment (timj) [inline docs/CHANGES.rst:2]: You have to shorten the `====`.

**Review by timj — COMMENTED** [review]
If you are going to drop the package name from the title you need to edit the template in pyproject.toml I think.

**Pull request merged**